app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb

```diff
@@ -0,0 +1,60 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # CreateJwksFromHttpResponse command class is responsible to create jwks object from http response
+      CreateJwksFromHttpResponse ||= CommandClass.new(
+        dependencies: {
+          logger: Rails.logger,
+          jwk_set_class: JSON::JWK::Set
+        },
+        inputs: %i[http_response]
+      ) do
+        def call
+          validate_response_exists
+          validate_response_has_a_body
+          create_jwks_from_http_respnse
+        end
+
+        private
+
+        def validate_response_exists
+          raise Errors::Authentication::AuthnJwt::MissingHttpResponse if @http_response.blank?
+        end
+
+        def validate_response_has_a_body
+          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
+        end
+
+        def create_jwks_from_http_respnse
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingJwksFromHttpResponse.new)
+
+          response_body = @http_response.body
+          encoded_body = Base64.encode64(response_body)
+          jwks = parse_jwks_response(response_body, encoded_body)
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedJwks.new)
```

Comment:
can we move this to the end of `parse_jwks_response` just before we return the value? this way we don't need to set the `jwks` param

Reply:
Done